homeassistant/components/media_player/liveboxplaytv.py

```diff
@@ -76,19 +77,32 @@ def __init__(self, host, port, name):
         self._channel_list = {}
         self._current_channel = None
         self._current_program = None
+        self._media_duration = None
+        self._media_remaining_time = None
+        self._media_ = None
         self._media_image_url = None
+        self._media_last_updated = None
 
     @asyncio.coroutine
     def async_update(self):
         """Retrieve the latest data."""
+        import pyteleloisirs
         try:
             self._state = self.refresh_state()
             # Update current channel
             channel = self._client.channel
             if channel is not None:
-                self._current_program = yield from \
-                    self._client.async_get_current_program_name()
                 self._current_channel = channel
+                program = yield from \
+                    self._client.async_get_current_program()
+                if program:
+                    self._current_program = program.get('name')
+                    # Media progress info
+                    self._media_duration = \
+                        pyteleloisirs.get_program_duration(program)
+                    self._media_remaining_time = \
```

Comment:
Media position and media remaining time need to only change when the player actually changes state. See vlc for an example https://github.com/home-assistant/home-assistant/blob/08b0629eca6417a2bb1b6f3c805ff19208484e4d/homeassistant/components/media_player/vlc.py#L70-L72

Reply:
Okay, but the media duration only depends on the current time. So this is pretty much guaranteed to be updated every time.